cli/kubernetes/platform/minikube/minikube.go

```diff
@@ -0,0 +1,47 @@
+package minikube
+
+import (
+	"context"
+
+	"github.com/suse/carrier/cli/kubernetes/platform/generic"
+
+	"github.com/kyokomi/emoji"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Minikube represents the minikube kubernetes platform.
+type Minikube struct {
+	generic.Generic
+}
+
+// Describe returns information about the platform.
+func (m *Minikube) Describe() string {
+	return emoji.Sprintf(":anchor:Detected kubernetes platform: %s\n:earth_americas:ExternalIPs: %s\n:curly_loop:InternalIPs: %s", m.String(), m.ExternalIPs(), m.InternalIPs)
+}
+
+func (m *Minikube) String() string { return "minikube" }
+
+// Detect detects if it is a minikube platform.
+func (m *Minikube) Detect(kube *kubernetes.Clientset) bool {
+	nodes, err := kube.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return false
+	}
+	for _, n := range nodes.Items {
+		if n.Name == "minikube" {
```

Comment:
The name of the node might be not a good identifier because you start a different cluster with the name `minikube`.
Since `ProviderID` is not there a better alternative might be scanning the Label keys for `minikube` instead.

Reply:
Labels on the nodes with keys in the form of `minikube.k8s.io/*` are present at least from minikube 1.9.2 to latest (1.16).

So, one could do:

```
		l := n.GetLabels()
		if _, found := l["minikube.k8s.io/version"]; found {
			return true
		}
```

Reply:
And v 18.2 has the labels as well.
```
    minikube.k8s.io/commit: eb13446e786c9ef70cb0a9f85a633194e62396a1
    minikube.k8s.io/name: minikube
    minikube.k8s.io/updated_at: 2021_01_14T16_02_02_0700
    minikube.k8s.io/version: v1.8.2
```
Adding my vote to using labels as well.

Reply:
Done. Changed to node labels